pkg/crd/crds.go

```diff
@@ -81,6 +87,27 @@ func ApplyCRDs(dynamicClient *dynamic.DynamicClient, clientset *kubernetes.Clien
 	return nil
 }
 
+// Convert GroupVersionKind to GroupVersionResource
+func groupVersionKindToResource(clientset kubernetes.Interface, gvk schema.GroupVersionKind) (*schema.GroupVersionResource, error) {
+	resourceList, err := clientset.Discovery().ServerPreferredResources()
+	if err != nil {
+		// ignore the error caused by a stale API service
+		if !strings.Contains(err.Error(), UnableToRetrieveCompleteAPIListError) {
+			return nil, err
```

Comment:
This is erring too aggressively. The problem in `err` might not be relevant to the particular `gvk` of interest here. See #1466

Reply:
@MikeSpreitzer completely agree with you.
the code that was pulled from KS Light is returning errors as is without wrapping them. 
this should be addressed all over the code (except for new transport code which already wraps errors).
In this PR I wasn't addressing this issue, but only replaced the concrete clients with interfaces to allow testing.

Reply:
@nirrozenbaum please open an issue on that

Reply:
It's a little strange to introduce a new function groupVersionKindToResource  that includes a bad behavior on th error handling.   It's not old code you ignore in this PR , it's new code you introduce .

Reply:
created an issue for this problem:
https://github.com/kubestellar/kubestellar/issues/1677

@ezrasilvera please pay attention, this is not new code, this function was pulled from KubeFlex code. 
In the existing code before this PR, KS code is calling functions from KF and this is one of the functions.
I pulled the function here since KF function still uses `clientset *kubernetes.Clientset` and without pulling the function here this change is not mergable. 
as I wrote to Mike, I completely agree this is an issue that should be addressed all over the code that was pulled from KSLight. let's try to handle this in #1677

Reply:
My OP here is not about wrapping errors. It is about the code failing when it could succeed.

FYI: the code in KubeFlex was cloned from code in KubeStellar _before_ #1466 merged.

Reply:
@nirrozenbaum please add an issue to fix this specific error , to make sure only the relevant errors fail the function (others might be OK in term of the logic here).

Reply:
Let's merge it and push a different PR to fix this bug

Reply:
@MikeSpreitzer I misunderstood. initially I thought you mean errors are not wrapped and also not explain the problem correctly in some of the case.
getting back to your point - you are correct.

my intention in this PR was not to fix this error but only to replace the specific clients with interface to allow testability.
copying this function from KF here is a side effect since the specific client appeared there as well.
I agree with the general approach to merge this PR to allow unit-test and will create an issue for the specific error.

Reply:
Yes, you did.

@pdettori : is there still unnecessary failure in KubeFlex?

Reply:
The imported function was from an older version of KubeFlex (which was used in KSLight). I changed since then that function, so no issues there.

https://github.com/kubestellar/kubeflex/blob/2287f51e6346f6f2b05b00ee52eb415cd3784eed/pkg/util/unstructured.go#L47-L62

Reply:
@MikeSpreitzer @pdettori please let me know if you want me to change the function to the newer version Paolo pointed to.